modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -741,9 +751,76 @@ def map_axis_partitions(
             right=None,
             lengths=lengths,
             enumerate_partitions=enumerate_partitions,
+            apply_func_args=map_func_args,
+            apply_func_kwargs=map_func_kwargs,
             **kwargs,
         )
 
+    @classmethod
+    def map_partitions_splitting_by_column(
+        cls,
+        partitions,
+        column_splits,
+        map_func,
+        map_func_args=None,
+        map_func_kwargs=None,
+    ):
+        """
+        Combine several blocks by column into one virtual partition and apply “map_funk” to them.
+
+        Parameters
+        ----------
+        partitions : NumPy 2D array
+            Partitions of Modin Frame.
+        column_splits : int
+            The number of splits by column.
+        map_func : callable
+            Function to apply.
+        map_func_args : iterable, optional
+            Positional arguments for the 'map_func'.
+        map_func_kwargs : dict, optional
+            Keyword arguments for the 'map_func'.
+
+        Returns
+        -------
+        NumPy array
+            An array of new partitions for Modin Frame.
+        """
+        new_partitions = np.array(
+            [
+                cls.column_partitions(
+                    partitions[i : i + column_splits],
+                    # full_axis=False,
+                )
+                for i in range(
+                    0,
+                    partitions.shape[0],
+                    column_splits,
+                )
+            ]
+        )
+        preprocessed_map_func = cls.preprocess_func(map_func)
+        kw = {
+            "num_splits": column_splits,
+        }
+        return np.concatenate(
```

Comment:
Would it be possible to reuse the logic from  map_axis_partitions instead?

Reply:
No. We are combining several partitions into one virtual partition and after execution we need to convert it back to keep the previous partitioning. if we use `map_axis_partitions` it will merge all virtual partitions into one axis partition and will be executed as one remote task.